api/management-model/src/test/java/org/apache/polaris/core/admin/model/CatalogSerializationTest.java

```diff
@@ -0,0 +1,305 @@
+
+        assertEquals(
+            originalAws.getRoleArn(), deserializedAws.getRoleArn(), "Role ARN should match");
+      }
+
+      assertEquals(
+          original.getStorageConfigInfo().getAllowedLocations(),
+          deserialized.getStorageConfigInfo().getAllowedLocations(),
+          "Allowed locations should match");
+    } else {
+      assertNull(deserialized.getStorageConfigInfo(), "Storage config should be null");
+    }
+
+    return deserialized;
+  }
+
+  // Update testCatalogSerialization to use verifyRoundTrip
+  @Test
+  public void testCatalogSerialization() throws JsonProcessingException {
+    CatalogProperties properties = new CatalogProperties(TEST_LOCATION);
+
+    // Create AWS storage config with required roleArn
+    StorageConfigInfo storageConfig =
+        new AwsStorageConfigInfo(TEST_ROLE_ARN, StorageConfigInfo.StorageTypeEnum.S3);
+
+    Catalog catalog =
+        new Catalog(Catalog.TypeEnum.INTERNAL, TEST_CATALOG_NAME, properties, storageConfig);
+
+    verifyRoundTrip(catalog);
+  }
+
+  // Update testCatalogDeserialization
+  @Test
+  public void testCatalogDeserialization() throws JsonProcessingException {
+    StorageConfigInfo storageConfig =
+        new AwsStorageConfigInfo(TEST_ROLE_ARN, StorageConfigInfo.StorageTypeEnum.S3);
+    Catalog catalog =
+        new Catalog(
+            Catalog.TypeEnum.INTERNAL,
+            TEST_CATALOG_NAME,
+            new CatalogProperties(TEST_LOCATION),
+            storageConfig);
+    verifyRoundTrip(catalog);
+  }
+
+  // Update testCatalogWithNullFields - already using proper verification
+  @Test
+  public void testCatalogWithNullFields() throws JsonProcessingException {
+    Catalog catalog = new Catalog(Catalog.TypeEnum.INTERNAL, null, null, null);
+    verifyRoundTrip(catalog);
+  }
+
+  // Tests error handling for invalid JSON input during deserialization
+  @Test
+  public void testInvalidJsonDeserialization() {
+    String invalidJson = "{ invalid json }";
+    assertThrows(JsonProcessingException.class, () -> mapper.readValue(invalidJson, Catalog.class));
+  }
+
+  // Tests handling of empty string values in catalog fields
+  @Test
+  public void testCatalogWithEmptyFields() throws JsonProcessingException {
+    String json =
+        """
+        {
+          "type": "INTERNAL",
+          "name": "",
+          "properties": {
+            "default-base-location": ""
+          },
+          "storageConfigInfo": {
+            "storageType": "S3",
+            "roleArn": "arn:aws:iam::123456789012:role/empty",
+            "allowedLocations": []
+          }
+        }""";
+
+    Catalog catalog = mapper.readValue(json, Catalog.class);
+    assertEquals("", catalog.getName());
+    assertEquals("", catalog.getProperties().getDefaultBaseLocation());
+  }
+
+  // Update testSpecialCharacters
+  @Test
+  public void testSpecialCharacters() throws JsonProcessingException {
+    String specialName = "test\"catalog";
```

Comment:
Let's make it one of the input options to the parameterized test for `verifyRoundTrip()`